ext/drbbdup/drbbdup.dox

```diff
@@ -29,3 +29,107 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
  * DAMAGE.
  */
+
+/**
+***************************************************************************
+***************************************************************************
+\page page_drbbdup Basic Block Duplicator
+
+The \p drbbdup DynamoRIO Basic Block Duplicator Extension provides a mechanism
+to efficiently support different instrumentations of the same basic blocks.
+Although flushing is one viable approach to achieve this functionality,
+it unfortunately also incurs high overheads. Instead, drbbdup duplicates the
+code of the basic block to generate multiple copies and enables a client to
+instrument these copies with different code insertions.
+
+\image html drbbdup_image.svg
+
+The duplication of a basic block is done at
+drmgr's app2app stage. Copies are prepended with a START label,
+and appended with a jump instruction to direct control to the EXIT label.
+Note that the last code version does not need this jump.
+Each basic block copy is set to handle a case of instrumentation. A case is
+denoted as a unique pointer-sized value that is registered by the client. The last
```

Comment:
suggest: s/A case is denoted as/Each case is identified by/

Reply:
Done.